balancer-js/src/lib/utils/poolHelper.ts

```diff
@@ -1,42 +1,82 @@
 import { parseFixed } from '@ethersproject/bignumber';
+import { AddressZero } from '@ethersproject/constants';
 import { Pool } from '../../types';
-import { _computeScalingFactor } from '@/lib/utils/solidityMaths';
-import { AssetHelpers } from './assetHelpers';
+import {
+  SolidityMaths,
+  _computeScalingFactor,
+  _upscaleArray,
+} from '@/lib/utils/solidityMaths';
+import { AssetHelpers } from '@/lib/utils/assetHelpers';
 
 const AMP_PRECISION = 3; // number of decimals -> precision 1000
 
+interface ParsedPoolInfo {
+  parsedTokens: string[];
+  parsedDecimals: string[];
+  parsedBalances: string[];
+  parsedWeights: string[];
+  parsedPriceRates: string[];
+  parsedAmp: string;
+  parsedTotalShares: string;
+  parsedSwapFee: string;
```

Comment:
100%. I really want to refactor this. A lot can probably be stripped out too. I think this will be better done in a separate PR once the other pool types have been refactored.